src/libraries/System.Net.HttpListener/src/System/Net/Managed/HttpListenerRequest.Managed.cs

```diff
@@ -87,10 +87,8 @@ internal void SetRequestLine(string req)
             _method = parts[0];
             foreach (char c in _method)
             {
-                int ic = (int)c;
-
-                if ((ic >= 'A' && ic <= 'Z') ||
-                    (ic > 32 && c < 127 && c != '(' && c != ')' && c != '<' &&
+                if (char.IsAsciiLetterUpper(c) ||
+                    (c > 32 && c < 127 && c != '(' && c != ')' && c != '<' &&
```

Comment:
Isn't IsAsciiLetterUpper redundant with... uh... the rest of that?

Maybe it's an intentional short-circuit?

Reply:
I assume it's a short-circuit.  Didn't want to dive in to optimizing this as that would be a giant hole :)